Tue Mar 20 03:53:11 FET 2012  Adam Vogt <vogt.adam@gmail.com>
  * Bump version to 0.10.1
  
  Raising the X11 dependency while keeping the xmonad version the same leads to
  problems where cabal install uses the dependency versions following hackage,
  not what is installed.
diff --git a/xmonad-contrib.cabal b/xmonad-contrib.cabal
index 743bdc8..0fbf4ac 100644
--- a/xmonad-contrib.cabal
+++ b/xmonad-contrib.cabal
@@ -61,7 +61,7 @@ library
         extensions: ForeignFunctionInterface
         cpp-options: -DXFT
 
-    build-depends:      mtl >= 1 && < 3, unix, X11>=1.5.0.0 && < 1.6, xmonad>=0.10, xmonad<0.11, utf8-string
+    build-depends:      mtl >= 1 && < 3, unix, X11>=1.5.0.0 && < 1.7, xmonad>=0.10, xmonad<0.11, utf8-string
 
     if true
         ghc-options:    -fwarn-tabs -Wall
diff --git a/XMonad/Layout/Spiral.hs b/XMonad/Layout/Spiral.hs
index a485d95..5e388e2 100644
--- a/XMonad/Layout/Spiral.hs
+++ b/XMonad/Layout/Spiral.hs
@@ -26,7 +26,7 @@ module XMonad.Layout.Spiral (
                             ) where
 
 import Data.Ratio
-import XMonad
+import XMonad (LayoutClass(..), Rectangle(..), fromMessage, Resize(..))
 import XMonad.StackSet ( integrate )
 
 -- $usage
diff --git a/XMonad/Util/Paste.hs b/XMonad/Util/Paste.hs
index 0d78653..4b7fd3b 100644
--- a/XMonad/Util/Paste.hs
+++ b/XMonad/Util/Paste.hs
@@ -23,7 +23,8 @@ module XMonad.Util.Paste ( -- * Usage
                            where
 
 import XMonad (io, theRoot, withDisplay, X ())
-import Graphics.X11
+import Graphics.X11 (Window, shiftMask, KeyMask, stringToKeysym, KeySym, keysymToKeycode)
+import Graphics.X11 (allocaXEvent, keyPress, sendEvent, keyPressMask, keyRelease, keyReleaseMask)
 import Graphics.X11.Xlib.Extras (none, setEventType, setKeyEvent)
 import Control.Monad.Reader (asks)
 import XMonad.Operations (withFocused)
